### PR TITLE
Make guest data available on Booking event objects

### PR DESCRIPTION
**Pull request by cgewecke**
+ Adds a helper method that extracts the `guestData` param from transactions whose hashes are captured by the event logs.
+ Adds `guestData` field to the event objects returned by `BookingData` and `HotelEvents`
+ Lets the `guestData` param of the Book methods be a string, and hex encodes within the lib.
+ Updates the tests to validate the above.

### CODE REVIEW

File: libs/BookingData.js [hunk 1]
```diff
@@ -121,15 +121,23 @@ class BookingData {
...
-      events.forEach(event => bookings.push({
-        transactionHash: event.transactionHash,
-        blockNumber: event.blockNumber,
-        id: event.id,
-        from: event.returnValues.from,
-        unit: event.returnValues.unit,
-        fromDate: util.parseDate(event.returnValues.fromDay),
-        daysAmount: event.returnValues.daysAmount
-      }));
+      if (!events.length)
```
Comment (AugustoL) [inline libs/BookingData.js]: why not use forEach or map?

File: libs/util/misc.js [hunk 2]
```diff
@@ -139,6 +139,27 @@ function bytes32ToString(hex){
...
+ * @param  {String} hash    transaction hash, available on the `CallStarted` event
+ * @param  {Object} context execution context of the class ()
+ * @return {String}      plain text guest data. If this is JSON it will need to be parsed.
+ */
+async function getGuestData(hash, context){
+  let guestData;
+  let tx = await context.web3.eth.getTransaction(hash);
+  let method = abiDecoder.decodeMethod(tx.input);
+
+  if (method.name === 'approveData'){
```
Comment (AugustoL) [inline libs/util/misc.js]: why this if?

File: test/BookingData.js [hunk 3]
```diff
@@ -89,7 +89,7 @@ describe('BookingData', function() {
     const fromDate = new Date('10/10/2020');
     const daysAmount = 5;
     const price = 1;
-    const guestData = web3.utils.toHex('guestData');
+    const guestData = 'guestData';
```
Comment (AugustoL) [inline test/BookingData.js]: I think we can work with guestData as a JSON object, parse it to string and then encode it before send it.

```const guestData = {}```

File: libs/BookingData.js (hunk 1 above)
Comment (cgewecke) [reply libs/BookingData.js]: Yes, we added a statement in the loop that returns a promise. `forEach` and `map` take a function argument. This means that we would need to make that argument async and use `Promise.all` to resolve the results. See a discussion of this [here](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop) on StackExchange. Of the choices for/let seems a little  cleaner.

Thanks for highlighting this though, I removed three redundant array length checks in the update - the loop logic catches that edge case.

File: libs/util/misc.js (hunk 2 above)
Comment (cgewecke) [reply libs/util/misc.js]: There are two transaction cases:`bookWithLif` and `book`. 

In the first, the `beginCall`, which contains the `guestData` payload is wrapped in an `approveData` tx. In the second `beginCall` is made directly. If the tx is `approveData` we have to drill down into the call one level to extract the data.

File: test/BookingData.js (hunk 3 above)
Comment (cgewecke) [reply test/BookingData.js]: I'm going to hold off on this change and open a related encryption issue here.

Comment (cgewecke) [reply test/BookingData.js]: #60

**Pull request merged**